src/main.rs

```diff
@@ -196,26 +214,58 @@ fn main() {
         .cloned()
         .collect();
 
-    let mut cargo = Command::new(cargo_bin());
-
-    cargo.envs(std::env::vars_os());
-
-    cargo
-        .arg("test")
-        .args(first_args)
-        .arg("--")
-        .args(second_args);
+    // Build the final cargo test command.
+    let mut cargo = format!(
+        "{} {} {} -- {}",
+        cargo_bin(),
+        custom_cargo_command.as_deref().unwrap_or_else(|| "test"),
+        first_args
+            .iter()
+            .map(|x| x.to_string_lossy())
+            .fold(String::new(), |mut acc, f| {
+                acc.push(' ');
+                acc.push_str(&f);
+                acc
+            }),
+        second_args
+            .iter()
+            .map(|x| x.to_string_lossy())
+            .fold(String::new(), |mut acc, f| {
+                acc.push(' ');
+                acc.push_str(&f);
+                acc
+            }),
+    );
 
     // We do this weird shit here since cargo does not support running multiple
     // tests individually.
     for item in final_items {
-        cargo.args(["--skip", &item]);
+        cargo.push_str(" --skip ");
+        cargo.push_str(&item);
     }
 
-    cargo.arg("--exact");
+    cargo.push_str(" --exact");
+
+    let mut cmd = {
+        #[cfg(target_family = "windows")]
+        {
+            let mut cmd = Command::new("cmd");
+            cmd.arg("/c");
+            cmd
+        }
```

Comment:
I will test this part on Windows platform in a couple of days.

Reply:
On windows, qtest with and without watch doesn't work right now. Output is always "system cannot find the path specified".